erudit/core/subscription/backends.py

```diff
@@ -0,0 +1,20 @@
+from django.contrib.auth.backends import ModelBackend
+from django.core.exceptions import PermissionDenied
+
+from .models import IndividualAccount
+
+
+class LegacyBackend(ModelBackend):
+    """ Authenticate users against IndividualAccount with sha1 custom
+    encryption from old system.
+    """
+
+    def authenticate(self, username=None, password=None):
+
+        try:
+            account = IndividualAccount.objects.get(username=username)
+        except IndividualAccount.DoesNotExist:
+            raise PermissionDenied()
```

Comment:
Si on utilise l'option one-to-one discutée plus bas, il ne faudrai pas faire de PermissionDenied ici, à moins que ce soit le dernier backend d'authentification. La BD d'abonnements va être uniquement utilisée pour authentifier l'utilisateur.